src/main/kotlin/org/rust/ide/newProject/RsDirectoryProjectGenerator.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.newProject
+
+import com.intellij.ide.util.projectWizard.AbstractNewProjectStep
+import com.intellij.ide.util.projectWizard.CustomStepProjectGenerator
+import com.intellij.ide.util.projectWizard.ProjectSettingsStepBase
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.wm.impl.welcomeScreen.AbstractActionWithPanel
+import com.intellij.platform.DirectoryProjectGenerator
+import com.intellij.platform.DirectoryProjectGeneratorBase
+import com.intellij.platform.ProjectGeneratorPeer
+import org.rust.ide.icons.RsIcons
+import javax.swing.Icon
+
+// We implement `CustomStepProjectGenerator` as well to correctly show settings UI
+// because otherwise PyCharm doesn't add peer's component into project settings panel
```

Comment:
Ah, so this is the magic ingredient to make it work in PyCharm! Ive pulled a lot of hair out over this one! Kudos!